### ISSUE TITLE
Route called even when middleware use redirection

### ISSUE THREAD

**Issue by snickers54** [alshdavid/crayon#43]
**Describe the bug**
Routes are still mounted even with a middleware that should prevent access to them (via a simple redirect). The redirection works well, but it shouldn't even access the component before the redirection is done. 

**To Reproduce**
1. Create a simple protected route with a console.log inside. Let's say /user
2. Use a custom middleware that simply redirect to another route. Let's say /
3. Access /user, you will see your console.log inside your protected route display in console.

**Expected behavior**
The route shouldn't mount or execute anything since there is a middleware and it purpose is to perform actions before accessing the actual path. 

**Desktop (please complete the following information):**
 - OS: MacOS
 - Crayon: ^2.0.5
 - Crayon svelte: ^1.0.1
 - Svelte: ^3.6.3
 - Typescript: ^3.5.3
 - Parcel plugin svelte: ^3.0.1
 - Parcel bundler: ^1.12.3

**Additional context**
main.js:
```javascript
import App from './App';
import crayon, { group } from 'crayon';
import svelte from 'crayon-svelte';
import Editor from './components/editor/Editor';
import Login from './components/auth/Login';

const app = crayon.create();
app.use(svelte.router());
const auth = group('/auth', group => {
    group.path('/login', async (req, res) => {
        await res.mount(App, {req, nav: app, component: Login});
    });
});

const secured = group('/bot');
secured.use((_, res) => {
    const session = localStorage.getItem('session');
    if (!session) {
        res.redirect('/auth/login');
    }
});
secured.path('/editor', async (req, res) => {
    await res.mount(App, {req, nav: app, component: Editor});
});
app.use(secured);
app.use(auth);
app.load();
```

App.svelte:
```javascript
<script>
    export let component;
    export let req;
    export let nav;
</script>

<div>
     <svelte:component this={component} nav={nav} req={req}/>
</div>
```

Editor.svelte:
```javascript
<script>
console.log("should not display");
</script>
```

Login.svelte:
```javascript
<div>login page</div>
```

**Comment by aigr** [comment]
+1

**Comment by alshdavid** [comment]
Thanks for noticing this, have put together [this PR](https://github.com/alshdavid/crayon/pull/46) to address it.

I have included two test cases to cover your this behaviour.

Will be available from `crayon` version `4.1.6`
Feel free to reopen this issue if the problem persists
